### ISSUE TITLE
[sklearn#34164] UI Collapse progress bars once completed?

### ISSUE THREAD

**Issue by godofecht** [flooooooooooow/flow#346]
**Mirrored from [scikit-learn#34164](https://github.com/scikit-learn/scikit-learn/issues/34164)**

Original title: UI Collapse progress bars once completed?
Original labels: Callbacks
Created: 2026-06-01T08:07:39Z

---

> [!WARNING]
> This issue is not yet ready for a PR. If you are interested in contributing to scikit-learn, please have a look at our [contributing guidelines](https://scikit-learn.org/dev/developers/contributing.html), and in particular the sections for [new contributors](https://scikit-learn.org/dev/developers/contributing.html#new-contributors) and the ["Needs triage"](https://scikit-learn.org/dev/developers/contributing.html#issues-tagged-needs-triage) label.

Right now `GridSearchCV` with callbacks end up using a lot of space in people's notebooks/ipython sessions after after they've finished running.

For example, this is what it looks like in ipython

<img width="931" height="391" alt="Image" src="https://github.com/user-attachments/assets/4c0a6c56-a291-4590-889e-7c5972a92409" />

This is from the proposed release highlights

<img width="768" height="661" alt="Image" src="https://github.com/user-attachments/assets/227cec58-26db-4d9a-8639-936a51d17099" />

I'm not sure how much value this adds once the run has completed. During the run you want to see this level of detail, but once it is done I'm not sure you need to see the progress bar for each of the folds for each of the parameters.

My proposal/request: can we add some shell/text output magic that "erases" the detailed progress bar output and replaces it with a single bar that is at 100% (or some other kind of summary line)? I think technically this is possible, but I don't have enough experience to know how fragile and complicated it would be.

What do you think?

---

## Flow context

This issue is mirrored from scikit-learn to explore whether Flow encounters the same problem or can approach it differently. Add analysis below.

**Comment by godofecht** [comment]
Moved to https://github.com/godofecht/flow-scikit/issues/77 in the flow-scikit repository. Further library-specific triage and implementation will happen there.
